release-notes/10.0/preview/preview1/libraries.md

```diff
@@ -8,7 +8,173 @@
+    {
+        // New overloads
+        public static int GetWeekOfYear(DateOnly date);
+        public static int GetYear(DateOnly date);
+        public static DateOnly ToDateOnly(int year, int week, DayOfWeek dayOfWeek);
+    }
+```
+
+## String Normalization APIs to Work with Span of Characters
+
+Unicode string normalization has been supported for a long time, but existing APIs have only worked with the string type. This means that callers with data stored in different forms, such as character arrays or spans, must allocate a new string to use these APIs.
+Additionally, APIs that return a normalized string always allocate a new string to represent the normalized output.
+
+The change introduces new APIs that work with spans of characters, reducing the restriction to string types and helping to avoid unnecessary allocations.
+
+```C#
+    public static class StringNormalizationExtensions
+    {
+        public static int GetNormalizedLength(this ReadOnlySpan<char> source, NormalizationForm normalizationForm = NormalizationForm.FormC);
+        public static bool IsNormalized(this ReadOnlySpan<char> source, NormalizationForm normalizationForm = NormalizationForm.FormC);
+        public static bool TryNormalize(this ReadOnlySpan<char> source, Span<char> destination, out int charsWritten, NormalizationForm normalizationForm = NormalizationForm.FormC);
+    }
+```
+
+## Adding TimeSpan.FromMilliseconds Overload with a Single Parameter
+
+Previously, we introduced the following method without adding an overload that takes a single parameter:
+
+```C#
+public static TimeSpan FromMilliseconds(long milliseconds, long microseconds = 0);
+```
+
+Although this works since the second parameter is optional, it causes a compilation error when used in a LINQ expression like:
+
+```C#
+Expression<Action> a = () => TimeSpan.FromMilliseconds(1000);
+```
+
+The issue arises because LINQ expressions cannot handle optional parameters. To address this, we are introducing an overload that takes a single parameter and modifying the existing method to make the second parameter mandatory:
+
+```C#
+public readonly struct TimeSpan
+{
+    public static TimeSpan FromMilliseconds(long milliseconds, long microseconds); // Second parameter is no longer optional
+    public static TimeSpan FromMilliseconds(long milliseconds);  // New overload
+}
+```
+
+## ZipArchive performance and memory improvements
+
+Two significant PRs have been made by contributor @edwardneal in .NET 10 Preview 1 to improve the performance and memory usage of `ZipArchive`:
+
+- [#102704](https://github.com/dotnet/runtime/pull/102704) - This PR optimizes the way entries are written to a `ZipArchive` when in `Update` mode. Previously, all `ZipArchiveEntries` would be loaded into memory and rewritten, which could lead to high memory usage and performance bottlenecks. The optimization reduces memory usage and improves performance by avoiding the need to load all entries into memory.

```

Comment:
I prefer this format. The repo is (largely) implied in this file but not across the body of release notes. We should be explicit. We've also use the longer form in past years.
 
```suggestion
- [dotnet/runtime #102704](https://github.com/dotnet/runtime/pull/102704) - This PR optimizes the way entries are written to a `ZipArchive` when in `Update` mode. Previously, all `ZipArchiveEntries` would be loaded into memory and rewritten, which could lead to high memory usage and performance bottlenecks. The optimization reduces memory usage and improves performance by avoiding the need to load all entries into memory.

```